process-automation/latest/model/data-models.md

```diff
@@ -0,0 +1,194 @@
+
+4. Use an arrow to connect to the Rest connector object.
+
+5. Select the Rest connector object and then click the **Append task** icon.
+
+6. Select the second object and click the **Change type** icon and then select **User Task**.
+
+7. Ensure the User task is still selected and then click **Assignment** in the properties pane.
+
+8. Select **Identity** and find the user you want to assign the task to and then click **Assign**.
+
+7. Select the User task and click **Append EndEvent**.
+
+8. Click anywhere in the whitespace to de-select the tasks and then click **Edit Process Variables** in the properties pane.
+
+9. Click the **+** symbol and enter a name for the variable called 'Person'.
+
+10. Click the **Type** dropdown list and select Data Models and then select the person data model. Click **Update**.
+
+    You can see the properties of your data model in the Process variables window.
+
+11. Select the REST connector object and then from the **Action** dropdown list select **Get**.
+
+     You are calling a service using the REST connector.
+
+12. In the Output mapping section select the Process variable for the `restResult` field to be `person`.
+
+     This means your person variable will be populated with the result which means you can use the data within your person task.
+
+12. Select the User task again and create a new form from the properties menu and call it 'person' and then add a textbox to it.
+
+13. In the **Field Editor** of the textbox add a label called 'Name' and then click **Save**.
+
+    **Note:** You must add a text box for each of the fields you created when you made the data model and you must label them accordingly.
+
+    You now have a user task that is using a form called person and that form contains a checkbox called 'Name'.
+  
+14. Go back to the process and select the user task and then edit the **Process variable/Value** field of the parameter called 'Name' that is under input mapping.
+
+     **Note:** You must do this for all the fields you want on your form.
+
+15. On the **Edit variable mapping** window select **Expression** and add `${person.Name}` and then click **Update**.
+
+    You have added the value that the Name field will use on the form. You can use the auto-completion functionality to enter the expression.
+
+    **Note:** You must do this for all the fields you want on your form.
+
+You now have a data model that is being used by a process that is processing complex objects.
+
+### Using Ref in a data model
+
+You can also use references when constructing your data model. Using the person data model you created earlier as an example, you can use references in that data model of objects you may want to re-use. For example a reference called ID (which is used company wide as the ID of an employee), and might be constructed of a username, and position could be used in numerous data models.
```

Comment:
The referenced block you are creating can be used just inside the data model where it is defined, and it makes sense when you reuse that definition in several properties of the data model you are defining.

The Data model editor allows you to reference other data models in the project but not referencing a definition inside another data model